custom_components/birthdays/__init__.py

```diff
@@ -13,15 +13,20 @@
 
 _LOGGER = logging.getLogger(__name__)
 
+CONF_UNIQUE_ID = 'unique_id'
 CONF_NAME = 'name'
 CONF_DATE_OF_BIRTH = 'date_of_birth'
 CONF_ICON = 'icon'
+CONF_ATTRIBUTES = 'attributes'
+CONF_AGE_AT_NEXT_BIRTHDAY = 'age_at_next_birthday'
 DOMAIN = 'birthdays'
 
 BIRTHDAY_CONFIG_SCHEMA = vol.Schema({
-    vol.Required(CONF_NAME): cv.string,
-    vol.Required(CONF_DATE_OF_BIRTH): cv.date,
-    vol.Optional(CONF_ICON, default='mdi:cake'): cv.string,
+    vol.Optional(CONF_UNIQUE_ID, default = None): cv.string,
```

Comment:
According to [these HA docs](https://developers.home-assistant.io/docs/development_validation/) they say
> Don't use a default which is None like vol.Optional(CONF_SOMETHING, default=None): cv.string, set the default to default='' if required.

I think your solution with None is better, however, `vol.Optional(CONF_UNIQUE_ID, default = None): cv.string` raises an error if CONF_UNIQUE_ID is not present. It must instead be `vol.Optional(CONF_UNIQUE_ID): cv.string`